system/lib/compiler-rt/stack_limits.S

```diff
@@ -0,0 +1,73 @@
+.globl emscripten_stack_init
+.globl emscripten_stack_set_limits
+.globl emscripten_stack_get_free
+.globl emscripten_stack_get_base
+.globl emscripten_stack_get_end
+
+.globaltype __stack_pointer, i32
+
+# TODO(sbc): It would be nice if these we initialized directly
+# using i32.const rather than using the `emscripten_stack_init`
```

Comment:
is that not possible in the .s syntax, or why are we not actually doing it?

Reply:
Yeah, we have not way to initialize globals in the `.s` syntax yet.  Would be a cool addition.

Turns out I think we might not be able to use that here any way (at least not in the dylink case).

Reply:
Come to think of it, does initializing in-memory globals with passive segments (needed for threads) work already?

Reply:
Yup, that is all implicit and is working.   It is done in the wasm start function but is guarded so it only happens on main thread (its skipped for secondary threads since it would globals r/w globals if not).   That is unrelated to wasm globals though isn't it?  Or am I misunderstanding?

Reply:
Yes, sorry it is unrelated to wasm globals. Thinking about wasm globals just made me realize I didn't know how that was handled in the assembly language either.